src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java

```diff
@@ -299,31 +299,24 @@ public boolean addAll(Collection<? extends E> c) {
 
     @Override
     public boolean removeAll(Collection<?> c) {
-        int numFalses = 0;
-        for (final Object o : c) {
-            if (!remove(o)) numFalses++;
-        }
-        return numFalses > 0;
+         c.forEach(this::remove);
+        return false;
     }
 
     @Override
     public boolean retainAll(Collection<?> c) {
-        int numFalses = 0;
 
         for (int i = 0; i < size(); i++) {
             final int headSlot = (int) ((head.get() + i) & mask);
             if (!c.contains(buffer[headSlot])) {
-                if (!remove(buffer[headSlot])) {
-                    numFalses++;
-                } else {
+                if (remove(buffer[headSlot])) {
                     // backtrack one step, we just backed values up at this point
                     i--;
                 }
-
             }
         }
 
-        return numFalses > 0;
```

Comment:
Seems like a breaking change as the test will never succeed.      Did you run the unit tests for this change?  I'm shocked that they didn't fail?     If they didn't please let me know or submit a pr with test coverage for this use case.   We should certainly have coverage for this as it is an FAQ.